src/ORM/ArrayList.php

```diff
@@ -596,8 +588,7 @@ public function canFilterBy($by)
      * Find the first item of this list where the given key = value
      *
      * @param string $key
-     * @param string $value
-     * @return mixed
+     * @param mixed $value
```

Comment:
`@return T|null`?

Reply:
I'm not sure what you're asking, please be more verbose. If you're asking "Why does this not include `@return T|null`?" then the answer is that this method and its signature and phpdoc typehints are inherited from the `Filterable` interface - there's no need to repeat the return typehint here since it doesn't differ from what's already declared in the PHPDoc on the interface's method declaration.

This is the same for any other methods which inherit methods from an interface or superclass where the typehint doesn't need to be repeated.

Reply:
OK, I think you meant `SS_List` interface

This PR is a bit of a nightmare since it essentially has lots of unrelated refactoring in it (changing docblocks not related to generics)

Reply:
OK, I think you meant `SS_List` interface

Kind feels like like there's loads of refactoring in this PR unrelated to generics specifically.  Since it's a bit of nightmare validating all of this and because docblocks types are known to be innaccurate I'm going to largely just ignore alot of those types of docblock changes where things were removed

That said - doesn't that now mean the interface now has the wrong docblock type? i.e. `SS_List::find()` should be `@return T|null`

We don't need to be too concerned about semver since docblocks aren't real so we're not going to break anyones site by changing them

Reply:
> That said - doesn't that now mean the interface now has the wrong docblock type? i.e. `SS_List::find()` should be `@return T|null`

It is `@return T|null`: https://github.com/silverstripe/silverstripe-framework/pull/11108/files#diff-55afe76c21c7906f822298c339509216643b0b56d33607ecf43e93bff6617b34R78